gitpod-shared/src/features.ts

```diff
@@ -387,12 +387,36 @@ export async function registerWorkspaceTimeout(context: GitpodExtensionContext):
 			}
 		});
 		try {
-			const result = await context.gitpod.server.setWorkspaceTimeout(context.info.workspaceId, '180m');
-			if (result.resetTimeoutOnWorkspaces?.length > 0) {
-				vscode.window.showWarningMessage('Workspace timeout has been extended to three hours. This reset the workspace timeout for other workspaces.');
-			} else {
-				vscode.window.showInformationMessage('Workspace timeout has been extended to three hours.');
+			await context.gitpod.server.setWorkspaceTimeout(context.info.workspaceId, '180m');
+			vscode.window.showWarningMessage(`Workspace timeout has been extended to 180m.`);
+		} catch (err) {
+			vscode.window.showErrorMessage(`Cannot extend workspace timeout: ${err.toString()}`);
+		}
+	}));
+
+
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.setWorkspaceTimeout', async () => {
+		const timeout = await vscode.window.showInputBox({
+			placeHolder: 'Please input the timeout time, such as 30m, 1h, 2h, 3h',
+			validateInput: (value: string) => {
+				if (!WorkspaceTimeoutDuration.validate(value)) {
```

Comment:
Can we echo back the error from the server? This may be more helpful 

https://github.com/gitpod-io/gitpod/blob/880f186042cec508af8328395f5546cd618fea76/components/gitpod-protocol/src/gitpod-service.ts#L364-L383